sherpa/cpp_api/online-recognizer.cc

```diff
@@ -226,15 +255,19 @@ class OnlineRecognizer::OnlineRecognizerImpl {
 
         model_ = std::make_unique<OnlineConformerTransducerModel>(
             config.nn_model, left_context, right_context, chunk_size, device_);
-      } else {
-        std::string s =
-            "Support only the following models from icefall:\n"
-            "conv_emformer_transducer_stateless2\n"
-            "pruned_stateless_emformer_rnnt2\n"
-            "pruned_transducer_stateless{2,3,4,5}\n";
-        TORCH_CHECK(false, s);
+        is_supported = true;
       }
     }
+    if (!is_supported) {
```

Comment:
```suggestion
    if (!model_) {
```
and remove `is_supported`.